appium/lib/screen-object/accessors.rb

```diff
@@ -75,19 +75,52 @@ def button(name, locator)
         end
 
         # generates method for scrolling on the screen and click on the button.
-        # this should be used for iOS platform.
-        # scroll to the first element with exact target static text or name.
+        # scroll to the first element with locator and click.
         # this method will not return any value.
         # button(:login_button,"xpath~//UIButtonField")
         # def scroll_button
-        #  login_button_scroll # This will not return any value. It will scroll on the screen until object found and click
-        #                        on the object i.e. button. This is iOS specific method and should not be used for android application
-        # end
-        define_method("#{name}_scroll") do
+        #  scroll_down_to_click_login_button # This will not return any value. It will scroll on the screen until object found and click
+        #                        on the object i.e. button.
+        define_method("scroll_down_to_click_#{name}") do
           # direction = options[:direction] || 'down'
-          ScreenObject::AppElements::Button.new(locator).scroll_for_element_click
+          ScreenObject::AppElements::Button.new(locator).scroll_element_to_view_click
+        end
+
+        # generates method for scrolling on the screen and click on the button.
+        # scroll to the first element with locator and click.
+        # this method will not return any value.
+        # button(:login_button,"xpath~//UIButtonField")
+        # def scroll_button
+        #  scroll_up_to_click_login_button # This will not return any value. It will scroll on the screen until object found and click
+        #                        on the object i.e. button.
+        define_method("scroll_up_to_click_#{name}") do
```

Comment:
Same, you added it to only a few element type classes but now all.

Reply:
Add all scroll methods to `TextField` class as well. Scroll and click will put focus in the text field and could be really helpful.